diff -Nru wvdial-1.60.orig/Makefile wvdial-1.60/Makefile
--- wvdial-1.60.orig/Makefile	2007-07-21 11:24:37.000000000 +0200
+++ wvdial-1.60/Makefile	2007-07-21 11:42:43.000000000 +0200
@@ -1,25 +1,17 @@
 
-prefix=/usr/local
-WVSTREAMS_INC=
-WVSTREAMS_LIB=
+prefix=$(DESTDIR)/usr
+WVSTREAMS_INC=$(WVS)/include
+WVSTREAMS_LIB=$(WVS)
 WVSTREAMS_BIN=$(prefix)/bin
 WVSTREAMS_SRC=.
 
-PC_CFLAGS=$(shell pkg-config --cflags libwvstreams)
-ifeq ($(PC_CFLAGS),)
- $(error WvStreams does not appear to be installed)
-endif
-CPPFLAGS+=$(PC_CFLAGS)
-
-PC_LIBS=$(shell pkg-config --libs libwvstreams)
-ifeq ($(PC_LIBS),)
- $(error WvStreams does not appear to be installed)
-endif
-LIBS+=$(PC_LIBS)
+LIBS = $(WVS)/libwvstatic.a -lstdc++ -lcrypto -lssl
 
 BINDIR=${prefix}/bin
-MANDIR=${prefix}/share/man
-PPPDIR=/etc/ppp/peers
+MANDIR=${prefix}/man
+PPPDIR=$(DESTDIR)/etc/ppp/peers
+
+VERBOSE=1
 
 include wvrules.mk
 
@@ -30,7 +22,6 @@
 wvdial.a: wvdialer.o wvmodemscan.o wvpapchap.o wvdialbrain.o \
 	wvdialmon.o
 
-wvdial: LDFLAGS+=-luniconf
 
 wvdial wvdialconf papchaptest pppmon: wvdial.a
 
