# http://bugs.gentoo.org/show_bug.cgi?&id=271477

diff -Nru wvdial-1.60.orig/wvmodemscan.cc wvdial-1.60/wvmodemscan.cc
--- wvdial-1.60.orig/wvmodemscan.cc	2009-06-04 10:40:35.913426298 +0000
+++ wvdial-1.60/wvmodemscan.cc	2009-06-04 10:40:55.723426715 +0000
@@ -496,7 +496,7 @@
 }
 
 
-static int filesort(const void *_e1, const void *_e2)
+static int filesort(const dirent **_e1, const dirent **_e2)
 {
     dirent const * const *e1 = (dirent const * const *)_e1;
     dirent const * const *e2 = (dirent const * const *)_e2;
