diff -Nru wvdial-1.55.orig/Makefile wvdial-1.55/Makefile
--- wvdial-1.55.orig/Makefile	2006-02-12 19:22:27.000000000 +0100
+++ wvdial-1.55/Makefile	2006-02-12 20:55:49.000000000 +0100
@@ -1,30 +1,22 @@
 ifeq ($(TOPDIR),)
  TOPDIR=.
 
- prefix=/usr/local
- WVSTREAMS_INC=
- WVSTREAMS_LIB=
+ prefix=$(DESTDIR)/usr
+ WVSTREAMS_INC=$(WVS)/include
+ WVSTREAMS_LIB=$(WVS)
  WVSTREAMS_BIN=$(prefix)/bin
  WVSTREAMS_SRC=.
 
- PC_CFLAGS=$(shell pkg-config --cflags libwvstreams)
- ifeq ($(PC_CFLAGS),)
-  $(error WvStreams does not appear to be installed)
- endif
- CPPFLAGS+=$(PC_CFLAGS)
-
- PC_LIBS=$(shell pkg-config --libs libwvstreams)
- ifeq ($(PC_LIBS),)
-  $(error WvStreams does not appear to be installed)
- endif
- LIBS+=$(PC_LIBS)
+ LIBS = $(WVS)/libuniconf.a $(WVS)/libwvstreams.a $(WVS)/libwvutils.a $(WVS)/libwvbase.a $(WVS)/xplc/libxplc-cxx.a -lcrypto -lssl
 else
  XPATH=$(TOPDIR)/src
 endif
 
 BINDIR=${prefix}/bin
-MANDIR=${prefix}/share/man
-PPPDIR=/etc/ppp/peers
+MANDIR=${prefix}/man
+PPPDIR=$(DESTDIR)/etc/ppp/peers
+
+VERBOSE=1
 
 include $(TOPDIR)/wvrules.mk
 
@@ -38,7 +30,7 @@
 ifeq ($(WVSTREAMS_LIB),)
  LIBUNICONF:=-luniconf
 endif
-wvdial wvdialconf papchaptest pppmon: wvdial.a $(LIBUNICONF)
+wvdial wvdialconf papchaptest pppmon: wvdial.a 
 
 install-bin: all
 	[ -d ${BINDIR}      ] || install -d ${BINDIR}
