--- calc-2.12.0.orig/Makefile	2006-05-19 17:00:01.000000000 +0300
+++ calc-2.12.0/Makefile	2006-05-21 11:26:59.000000000 +0300
@@ -637,7 +637,7 @@
 CUSTOMCALDIR= ${CALC_SHAREDIR}/custom
 CUSTOMHELPDIR= ${CALC_SHAREDIR}/custhelp
 CUSTOMINCDIR= ${CALC_INCDIR}/custom
-SCRIPTDIR= ${BINDIR}/cscript
+SCRIPTDIR= /usr/lib/calc
 
 # T - top level directory under which calc will be installed
 #
@@ -785,11 +785,11 @@
 #
 # If in doubt, set USE_READLINE, READLINE_LIB and READLINE_INCLUDE to nothing.
 #
-USE_READLINE=
-#USE_READLINE= -DUSE_READLINE
+#USE_READLINE=
+USE_READLINE= -DUSE_READLINE
 #
-READLINE_LIB=
-#READLINE_LIB= -lreadline -lhistory -lncurses
+#READLINE_LIB=
+READLINE_LIB= -lreadline -lhistory -lncurses
 #READLINE_LIB= -L/usr/gnu/lib -lreadline -lhistory -lncurses
 #READLINE_LIB= -L/usr/local/lib -lreadline -lhistory -lncurses
 #
@@ -834,7 +834,7 @@
 #
 #DEBUG= -O3
 #DEBUG= -O3 -g
-DEBUG= -O3 -g3
+#DEBUG= -O3 -g3
 #DEBUG= -O3 -ipa
 #DEBUG= -O3 -g3 -ipa
 #
