diff -Nru fcron-3.0.1.orig/configure.in fcron-3.0.1/configure.in
--- fcron-3.0.1.orig/configure.in	2006-02-07 18:02:49.000000000 +0100
+++ fcron-3.0.1/configure.in	2006-02-07 18:03:02.000000000 +0100
@@ -268,13 +268,8 @@
     fi
     ;;
   *)
-    if test -d "$withval"; then
-      etcdir="$withval"
-      AC_MSG_RESULT($withval)
-    else
-      AC_MSG_ERROR([
-Directory $withval does not exist])
-    fi
+    etcdir="$withval"
+    AC_MSG_RESULT($withval)
     ;;
   esac ],
   if test -d "/etc"; then
