diff -Nru fcron-3.0.2.orig/configure.in fcron-3.0.2/configure.in
--- fcron-3.0.2.orig/configure.in	2007-01-17 08:54:11.000000000 +0100
+++ fcron-3.0.2/configure.in	2007-01-17 08:54:23.000000000 +0100
@@ -278,13 +278,8 @@
     fi
     ;;
   *)
-    if test -d "$withval"; then
-      etcdir="$withval"
-      AC_MSG_RESULT($withval)
-    else
-      AC_MSG_ERROR([
-Directory $withval does not exist])
-    fi
+    etcdir="$withval"
+    AC_MSG_RESULT($withval)
     ;;
   esac ],
   if test -d "/etc"; then
