diff -Nru dnsmasq-2.24.orig/Makefile dnsmasq-2.24/Makefile
--- dnsmasq-2.24.orig/Makefile	2005-11-27 09:54:12.000000000 +0100
+++ dnsmasq-2.24/Makefile	2005-11-27 09:55:51.000000000 +0100
@@ -1,6 +1,6 @@
 PREFIX?=/usr/local
 BINDIR = ${PREFIX}/sbin
-MANDIR = ${PREFIX}/share/man
+MANDIR = ${PREFIX}/man
 LOCALEDIR = ${PREFIX}/share/locale
 
 SRC = src
diff -Nru dnsmasq-2.24.orig/dnsmasq.conf.example dnsmasq-2.24/dnsmasq.conf.example
--- dnsmasq-2.24.orig/dnsmasq.conf.example	2005-11-27 09:54:12.000000000 +0100
+++ dnsmasq-2.24/dnsmasq.conf.example	2005-11-27 09:55:08.000000000 +0100
@@ -59,8 +59,8 @@
 
 # If you want dnsmasq to change uid and gid to something other
 # than the default, edit the following lines.
-#user=
-#group=
+user=nobody
+group=nobody
 
 # If you want dnsmasq to listen for DHCP and DNS requests only on
 # specified interfaces (and the loopback) give the name of the
@@ -262,7 +262,7 @@
 # The DHCP server needs somewhere on disk to keep its lease database.
 # This defaults to a sane location, but if you want to change it, use
 # the line below.
-#dhcp-leasefile=/var/lib/misc/dnsmasq.leases
+# dhcp-leasefile=/var/lib/dhcp/dnsmasq.leases
 
 # Set the DHCP server to authoritative mode. In this mode it will barge in
 # and take over the lease for any client which broadcasts on the network,
diff -Nru dnsmasq-2.24.orig/src/config.h dnsmasq-2.24/src/config.h
--- dnsmasq-2.24.orig/src/config.h	2005-11-27 09:54:12.000000000 +0100
+++ dnsmasq-2.24/src/config.h	2005-11-27 09:55:08.000000000 +0100
@@ -35,7 +35,7 @@
 #if defined(__FreeBSD__) || defined (__OpenBSD__)
 #   define LEASEFILE "/var/db/dnsmasq.leases"
 #else
-#   define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#   define LEASEFILE "/var/lib/dhcp/dnsmasq.leases"
 #endif
 #if defined(__FreeBSD__)
 #   define CONFFILE "/usr/local/etc/dnsmasq.conf"
@@ -44,7 +44,7 @@
 #endif
 #define DEFLEASE 3600 /* default lease time, 1 hour */
 #define CHUSER "nobody"
-#define CHGRP "dip"
+#define CHGRP "nobody"
 #define IP6INTERFACES "/proc/net/if_inet6"
 #define UPTIME "/proc/uptime"
 #define DHCP_SERVER_PORT 67
