diff -Nru dnsmasq-2.47.orig/Makefile dnsmasq-2.47/Makefile
--- dnsmasq-2.47.orig/Makefile	2009-02-06 09:20:22.000000000 +0100
+++ dnsmasq-2.47/Makefile	2009-02-06 09:21:25.000000000 +0100
@@ -15,7 +15,7 @@
 
 PREFIX = /usr/local
 BINDIR = ${PREFIX}/sbin
-MANDIR = ${PREFIX}/share/man
+MANDIR = ${PREFIX}/man
 LOCALEDIR = ${PREFIX}/share/locale
 
 SRC = src
@@ -23,7 +23,7 @@
 MAN = man
 
 PKG_CONFIG = pkg-config
-AWK = nawk
+AWK = awk
 INSTALL = install
 
 DBUS_MINOR=" `echo $(COPTS) | ../bld/pkg-wrapper $(PKG_CONFIG) --modversion dbus-1 | $(AWK) -F . -- '{ if ($$(NF-1)) print \"-DDBUS_MINOR=\"$$(NF-1) }'`" 
diff -Nru dnsmasq-2.47.orig/bld/Makefile dnsmasq-2.47/bld/Makefile
--- dnsmasq-2.47.orig/bld/Makefile	2009-02-06 09:20:22.000000000 +0100
+++ dnsmasq-2.47/bld/Makefile	2009-02-06 09:20:35.000000000 +0100
@@ -1,4 +1,4 @@
-CFLAGS = -Wall -W -O2
+CFLAGS += -Wall -W
 
 OBJS = cache.o rfc1035.o util.o option.o forward.o network.o \
        dnsmasq.o dhcp.o lease.o rfc2131.o netlink.o dbus.o bpf.o \
diff -Nru dnsmasq-2.47.orig/dnsmasq.conf.example dnsmasq-2.47/dnsmasq.conf.example
--- dnsmasq-2.47.orig/dnsmasq.conf.example	2009-02-06 09:20:22.000000000 +0100
+++ dnsmasq-2.47/dnsmasq.conf.example	2009-02-06 09:20:36.000000000 +0100
@@ -75,8 +75,8 @@
 
 # If you want dnsmasq to change uid and gid to something other
 # than the default, edit the following lines.
-#user=
-#group=
+#user=nobody
+#group=nobody
 
 # If you want dnsmasq to listen for DHCP and DNS requests only on
 # specified interfaces (and the loopback) give the name of the
@@ -381,7 +381,7 @@
 # The DHCP server needs somewhere on disk to keep its lease database.
 # This defaults to a sane location, but if you want to change it, use
 # the line below.
-#dhcp-leasefile=/var/lib/misc/dnsmasq.leases
+# dhcp-leasefile=/var/lib/dhcp/dnsmasq.leases
 
 # Set the DHCP server to authoritative mode. In this mode it will barge in
 # and take over the lease for any client which broadcasts on the network,
diff -Nru dnsmasq-2.47.orig/src/config.h dnsmasq-2.47/src/config.h
--- dnsmasq-2.47.orig/src/config.h	2009-02-06 09:20:22.000000000 +0100
+++ dnsmasq-2.47/src/config.h	2009-02-06 09:20:36.000000000 +0100
@@ -45,7 +45,7 @@
 #   elif defined(__sun__) || defined (__sun)
 #      define LEASEFILE "/var/cache/dnsmasq.leases"
 #   else
-#      define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#      define LEASEFILE "/var/lib/dhcp/dnsmasq.leases"
 #   endif
 #endif
 
@@ -59,7 +59,7 @@
 
 #define DEFLEASE 3600 /* default lease time, 1 hour */
 #define CHUSER "nobody"
-#define CHGRP "dip"
+#define CHGRP "nobody"
 #define DHCP_SERVER_PORT 67
 #define DHCP_CLIENT_PORT 68
 #define DHCP_SERVER_ALTPORT 1067
