diff -Nru dnsmasq-2.46.orig/Makefile dnsmasq-2.46/Makefile
--- dnsmasq-2.46.orig/Makefile	2008-11-14 23:43:19.000000000 +0100
+++ dnsmasq-2.46/Makefile	2008-11-14 23:43:35.000000000 +0100
@@ -15,7 +15,7 @@
 
 PREFIX = /usr/local
 BINDIR = ${PREFIX}/sbin
-MANDIR = ${PREFIX}/share/man
+MANDIR = ${PREFIX}/man
 LOCALEDIR = ${PREFIX}/share/locale
 
 SRC = src
diff -Nru dnsmasq-2.46.orig/bld/Makefile dnsmasq-2.46/bld/Makefile
--- dnsmasq-2.46.orig/bld/Makefile	2008-11-14 23:43:19.000000000 +0100
+++ dnsmasq-2.46/bld/Makefile	2008-11-14 23:43:35.000000000 +0100
@@ -1,4 +1,4 @@
-CFLAGS = -Wall -W -O2
+CFLAGS += -Wall -W
 
 OBJS = cache.o rfc1035.o util.o option.o forward.o network.o \
        dnsmasq.o dhcp.o lease.o rfc2131.o netlink.o dbus.o bpf.o \
diff -Nru dnsmasq-2.46.orig/dnsmasq.conf.example dnsmasq-2.46/dnsmasq.conf.example
--- dnsmasq-2.46.orig/dnsmasq.conf.example	2008-11-14 23:43:19.000000000 +0100
+++ dnsmasq-2.46/dnsmasq.conf.example	2008-11-14 23:43:35.000000000 +0100
@@ -75,8 +75,8 @@
 
 # If you want dnsmasq to change uid and gid to something other
 # than the default, edit the following lines.
-#user=
-#group=
+#user=nobody
+#group=nobody
 
 # If you want dnsmasq to listen for DHCP and DNS requests only on
 # specified interfaces (and the loopback) give the name of the
@@ -365,7 +365,7 @@
 # The DHCP server needs somewhere on disk to keep its lease database.
 # This defaults to a sane location, but if you want to change it, use
 # the line below.
-#dhcp-leasefile=/var/lib/misc/dnsmasq.leases
+# dhcp-leasefile=/var/lib/dhcp/dnsmasq.leases
 
 # Set the DHCP server to authoritative mode. In this mode it will barge in
 # and take over the lease for any client which broadcasts on the network,
diff -Nru dnsmasq-2.46.orig/src/config.h dnsmasq-2.46/src/config.h
--- dnsmasq-2.46.orig/src/config.h	2008-11-14 23:43:19.000000000 +0100
+++ dnsmasq-2.46/src/config.h	2008-11-14 23:43:35.000000000 +0100
@@ -43,7 +43,7 @@
 #elif defined(__sun__) || defined (__sun)
 #   define LEASEFILE "/var/cache/dnsmasq.leases"
 #else
-#   define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#   define LEASEFILE "/var/lib/dhcp/dnsmasq.leases"
 #endif
 #if defined(__FreeBSD__)
 #   define CONFFILE "/usr/local/etc/dnsmasq.conf"
@@ -52,7 +52,7 @@
 #endif
 #define DEFLEASE 3600 /* default lease time, 1 hour */
 #define CHUSER "nobody"
-#define CHGRP "dip"
+#define CHGRP "nobody"
 #define DHCP_SERVER_PORT 67
 #define DHCP_CLIENT_PORT 68
 #define DHCP_SERVER_ALTPORT 1067
