diff -Nru clex-3.15_test1.orig/src/inout.c clex-3.15_test1/src/inout.c
--- clex-3.15_test1.orig/src/inout.c	2006-05-10 20:54:10.000000000 +0200
+++ clex-3.15_test1/src/inout.c	2006-05-10 20:55:42.000000000 +0200
@@ -290,7 +290,7 @@
 		if (key == KEY_SUSPEND)
 			return CH_CTRL('Z');
 #endif
-		if (key == KEY_BACKSPACE)
+		if (key == KEY_BACKSPACE || key == 127)
 			return CH_CTRL('H');
 		if (key == KEY_ENTER)
 			return CH_CTRL('M');
