--- wmii-2.5.2/cmd/wm/page.c	2006-01-27 22:09:37.000000000 +0100
+++ wmii-2.5.2-new/cmd/wm/page.c	2006-04-18 23:04:07.000000000 +0200
@@ -275,9 +275,6 @@
 	XTranslateCoordinates(dpy, f->win, root, wex, wey, &ex, &ey, &dummy);
 	if (blitz_ispointinrect(ex, ey, r))
 		return;
-	XSelectInput(dpy, root, ROOT_MASK & ~EnterWindowMask);
-	XWarpPointer(dpy, None, f->win, 0, 0, 0, 0, r->width / 2, r->height / 2);
-	XSelectInput(dpy, root, ROOT_MASK);
 }
 
 static void 
