diff -Nru vsftpd-2.2.1.orig/Makefile vsftpd-2.2.1/Makefile
--- vsftpd-2.2.1.orig/Makefile	2009-10-19 13:05:15.000000000 +0200
+++ vsftpd-2.2.1/Makefile	2009-10-19 13:05:28.000000000 +0200
@@ -3,7 +3,7 @@
 INSTALL	=	install
 IFLAGS  = -idirafter dummyinc
 #CFLAGS = -g
-CFLAGS	=	-O2 -Wall -W -Wshadow #-pedantic -Werror -Wconversion
+CFLAGS	+=	-Wall -W -Wshadow #-pedantic -Werror -Wconversion
 
 LIBS	=	`./vsf_findlibs.sh`
 LINK	=	-Wl,-s
@@ -24,21 +24,10 @@
 	$(CC) -o vsftpd $(OBJS) $(LINK) $(LIBS) $(LDFLAGS)
 
 install:
-	if [ -x /usr/local/sbin ]; then \
-		$(INSTALL) -m 755 vsftpd /usr/local/sbin/vsftpd; \
-	else \
-		$(INSTALL) -m 755 vsftpd /usr/sbin/vsftpd; fi
-	if [ -x /usr/local/man ]; then \
-		$(INSTALL) -m 644 vsftpd.8 /usr/local/man/man8/vsftpd.8; \
-		$(INSTALL) -m 644 vsftpd.conf.5 /usr/local/man/man5/vsftpd.conf.5; \
-	elif [ -x /usr/share/man ]; then \
-		$(INSTALL) -m 644 vsftpd.8 /usr/share/man/man8/vsftpd.8; \
-		$(INSTALL) -m 644 vsftpd.conf.5 /usr/share/man/man5/vsftpd.conf.5; \
-	else \
-		$(INSTALL) -m 644 vsftpd.8 /usr/man/man8/vsftpd.8; \
-		$(INSTALL) -m 644 vsftpd.conf.5 /usr/man/man5/vsftpd.conf.5; fi
-	if [ -x /etc/xinetd.d ]; then \
-		$(INSTALL) -m 644 xinetd.d/vsftpd /etc/xinetd.d/vsftpd; fi
+	$(INSTALL) -D -m 755 vsftpd $(DESTDIR)/usr/sbin/vsftpd
+	$(INSTALL) -D -m 644 vsftpd.8 $(DESTDIR)/usr/man/man8/vsftpd.8
+	$(INSTALL) -D -m 644 vsftpd.conf.5 $(DESTDIR)/usr/man/man5/vsftpd.conf.5
+	$(INSTALL) -D -m 600 vsftpd.conf $(DESTDIR)/etc/vsftpd.conf
 
 clean:
 	rm -f *.o *.swp vsftpd
diff -Nru vsftpd-2.2.1.orig/builddefs.h vsftpd-2.2.1/builddefs.h
--- vsftpd-2.2.1.orig/builddefs.h	2009-10-19 13:05:15.000000000 +0200
+++ vsftpd-2.2.1/builddefs.h	2009-10-19 13:05:28.000000000 +0200
@@ -1,9 +1,9 @@
 #ifndef VSF_BUILDDEFS_H
 #define VSF_BUILDDEFS_H
 
-#undef VSF_BUILD_TCPWRAPPERS
-#define VSF_BUILD_PAM
-#undef VSF_BUILD_SSL
+#define VSF_BUILD_TCPWRAPPERS
+#undef VSF_BUILD_PAM
+#define VSF_BUILD_SSL
 
 #endif /* VSF_BUILDDEFS_H */
 
diff -Nru vsftpd-2.2.1.orig/tunables.c vsftpd-2.2.1/tunables.c
--- vsftpd-2.2.1.orig/tunables.c	2009-10-19 13:05:15.000000000 +0200
+++ vsftpd-2.2.1/tunables.c	2009-10-19 13:05:28.000000000 +0200
@@ -246,7 +246,7 @@
   /* -rw------- */
   tunable_chown_upload_mode = 0600;
 
-  install_str_setting("/usr/share/empty", &tunable_secure_chroot_dir);
+  install_str_setting("/var/empty", &tunable_secure_chroot_dir);
   install_str_setting("ftp", &tunable_ftp_username);
   install_str_setting("root", &tunable_chown_username);
   install_str_setting("/var/log/xferlog", &tunable_xferlog_file);
@@ -273,11 +273,11 @@
   install_str_setting(0, &tunable_user_sub_token);
   install_str_setting("/etc/vsftpd.email_passwords",
                       &tunable_email_password_file);
-  install_str_setting("/usr/share/ssl/certs/vsftpd.pem",
+  install_str_setting("/etc/ssl/certs/vsftpd.crt",
                       &tunable_rsa_cert_file);
   install_str_setting(0, &tunable_dsa_cert_file);
   install_str_setting("DES-CBC3-SHA", &tunable_ssl_ciphers);
-  install_str_setting(0, &tunable_rsa_private_key_file);
+  install_str_setting("/etc/ssl/keys/vsftpd.key", &tunable_rsa_private_key_file);
   install_str_setting(0, &tunable_dsa_private_key_file);
   install_str_setting(0, &tunable_ca_certs_file);
 }
diff -Nru vsftpd-2.2.1.orig/vsftpd.conf vsftpd-2.2.1/vsftpd.conf
--- vsftpd-2.2.1.orig/vsftpd.conf	2009-10-19 13:05:15.000000000 +0200
+++ vsftpd-2.2.1/vsftpd.conf	2009-10-19 13:05:28.000000000 +0200
@@ -8,11 +8,25 @@
 # Please read the vsftpd.conf.5 manual page to get a full idea of vsftpd's
 # capabilities.
 #
+#
+# Run standalone
+listen=YES
+background=YES
+#
+# Use tcp_wrappers library
+tcp_wrappers=YES
+#
+# Enable SSL support
+#ssl_enable=YES
+#ssl_sslv3=YES
+#
 # Allow anonymous FTP? (Beware - allowed by default if you comment this out).
 anonymous_enable=YES
 #
 # Uncomment this to allow local users to log in.
 #local_enable=YES
+#force_local_logins_ssl=YES
+#force_local_data_ssl=YES
 #
 # Uncomment this to enable any form of FTP write command.
 #write_enable=YES
diff -Nru vsftpd-2.2.1.orig/vsftpd.conf.5 vsftpd-2.2.1/vsftpd.conf.5
--- vsftpd-2.2.1.orig/vsftpd.conf.5	2009-10-19 13:05:15.000000000 +0200
+++ vsftpd-2.2.1/vsftpd.conf.5	2009-10-19 13:05:28.000000000 +0200
@@ -955,21 +955,21 @@
 This option specifies the location of the RSA certificate to use for SSL
 encrypted connections.
 
-Default: /usr/share/ssl/certs/vsftpd.pem
+Default: /etc/ssl/certs/vsftpd.crt
 .TP
 .B rsa_private_key_file
 This option specifies the location of the RSA private key to use for SSL
 encrypted connections. If this option is not set, the private key is expected
 to be in the same file as the certificate.
 
-Default: (none)
+Default: /etc/ssl/keys/vsftpd.key
 .TP
 .B secure_chroot_dir
 This option should be the name of a directory which is empty. Also, the
 directory should not be writable by the ftp user. This directory is used
 as a secure chroot() jail at times vsftpd does not require filesystem access.
 
-Default: /usr/share/empty
+Default: /var/empty
 .TP
 .B ssl_ciphers
 This option can be used to select which SSL ciphers vsftpd will allow for
