--- pkgfoster	2006-12-19 17:52:33.000000000 +0100
+++ /home/simone/crux/tools/prt-utils/pkgfoster	2007-05-15 18:05:12.464471239 +0200
@@ -12,9 +12,22 @@
 #
 
 PRT_GET=prt-cache
-PKGRM="sudo pkgrm"
-BASE="autoconf automake bash bc bin86 bindutils binutils bison bzip2 coreutils cpio curl db dcron dhcpcd diffutils e2fsprogs ed expat file filesystem findutils flex gawk gcc gdbm gettext glibc grep groff grub gzip hdparm hotplug httpup iptables jfsutils kbd less libstdc++-compat libtool libusb lilo m4 make man man-pages mktemp module-init-tools nasm ncurses netkit-base netkit-ftp netkit-telnet net-tools nfs-utils openssh openssl patch pciutils perl pkg-config pkgutils portmap ports ppp procps prt-get psmisc rc rdate readline reiserfsprogs rpm2targz rsync sed sendmail shadow slocate strace sysfsutils sysklogd sysvinit tar tcp_wrappers tcsh time traceroute udev unzip usbutils util-linux vim wget which xfsprogs zip zlib"
 
+function PKGRM
+{
+	if [ $UID -ne 0 ]; then
+		SUDO="`which sudo 2>/dev/null`"
+		if [ -z "$SUDO" ]; then
+			su -c "pkgrm $1"
+		else
+			sudo pkgrm $1
+		fi
+	else
+		pkgrm $1
+	fi
+}
+
+BASE=$(ls $(awk '/^[[:space:]]*prtdir.*\/core/ {print $2}' /etc/prt-get.conf))
 touch ~/.keepers
 
 echo "Checking packages for orphans..."
@@ -37,7 +50,7 @@
 			read ANSWER
 
 			if [ "$ANSWER" == "y" ] ; then
-				$PKGRM $PACKAGE
+				PKGRM $PACKAGE
 				RECHECK=1
 			else
 				echo $PACKAGE >> ~/.keepers
