--- imlib2-1.2.2.001/configure.org	2006-03-31 19:33:20.000000000 +0200
+++ imlib2-1.2.2.001/configure	2006-03-31 19:34:17.000000000 +0200
@@ -20451,7 +20451,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lungif  $LIBS"
+LIBS="-lungif ­L/usr/X11R6/lib -lX11 -lSM -lICE $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -20509,7 +20509,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_ungif_DGifOpenFileName" >&5
 echo "${ECHO_T}$ac_cv_lib_ungif_DGifOpenFileName" >&6
 if test $ac_cv_lib_ungif_DGifOpenFileName = yes; then
-  gif_libs="-lungif"
+  gif_libs="-lungif -L/usr/X11R6/lib -X11 -lSM -lICE"
       gif_ok=yes
 else
   gif_ok=no
