diff -Nru pcmcia-cs-3.2.7/Makefile pcmcia-cs-3.2.7-new/Makefile
--- pcmcia-cs-3.2.7/Makefile	2003-05-16 07:59:44.000000000 +0200
+++ pcmcia-cs-3.2.7-new/Makefile	2004-05-01 17:34:00.837318400 +0200
@@ -6,11 +6,8 @@
 include config.mk
 endif
 
-ALL  = modules clients wireless cardmgr flash debug-tools man etc
-DIRS = cardmgr flash debug-tools man etc
-ifndef CONFIG_PCMCIA
-DIRS := modules clients wireless $(DIRS)
-endif
+ALL  = cardmgr debug-tools man etc
+DIRS = cardmgr debug-tools man etc
 
 help:
 	@echo "Pick one of the following targets:"
@@ -31,7 +28,7 @@
 	@$(MAKE) -s clean
 	@./Configure -n
 
-all:	.prereq.ok kcheck
+all:
 	@set -e ; for d in $(DIRS) ; do $(MAKE) -C $$d ; done
 	@for f in *.mk ; do \
 	    if [ $$f != config.mk -a $$f != rules.mk ] ; then \
@@ -48,7 +45,7 @@
 	rm -f config.out
 	@$(MAKE) clean
 
-install: .prereq.ok kcheck
+install:
 	@set -e ; for d in $(DIRS) ; do $(MAKE) -C $$d install ; done
 	@for f in *.mk ; do \
 	    if [ $$f != config.mk -a $$f != rules.mk ] ; then \
diff -Nru pcmcia-cs-3.2.7/cardmgr/Makefile pcmcia-cs-3.2.7-new/cardmgr/Makefile
--- pcmcia-cs-3.2.7/cardmgr/Makefile	2003-11-22 09:54:13.000000000 +0100
+++ pcmcia-cs-3.2.7-new/cardmgr/Makefile	2004-05-01 18:01:13.602100424 +0200
@@ -110,7 +110,7 @@
 install: $(TOOLS) $(I_EXTRA)
 	@mkdir -p $(PREFIX)/sbin
 	@rm -f $(PREFIX)/sbin/cardmgr
-	cp -f $(TOOLS) pcinitrd $(PREFIX)/sbin
+	cp -f $(TOOLS) $(PREFIX)/sbin
 	chmod u+s $(PREFIX)/sbin/cardctl
 
 include ../rules.mk
diff -Nru pcmcia-cs-3.2.7/config.mk pcmcia-cs-3.2.7-new/config.mk
--- pcmcia-cs-3.2.7/config.mk	1970-01-01 01:00:00.000000000 +0100
+++ pcmcia-cs-3.2.7-new/config.mk	2004-05-01 17:34:27.748227320 +0200
@@ -0,0 +1,7 @@
+LINUX=/usr
+PREFIX=/
+UCC=gcc
+LD=ld
+UFLAGS=
+CPPFLAGS=-I/usr/include -I../include
+MANDIR=/usr/man
diff -Nru pcmcia-cs-3.2.7/etc/install-etc pcmcia-cs-3.2.7-new/etc/install-etc
--- pcmcia-cs-3.2.7/etc/install-etc	2002-07-17 08:41:49.000000000 +0200
+++ pcmcia-cs-3.2.7-new/etc/install-etc	2004-05-01 17:35:18.188559224 +0200
@@ -7,7 +7,7 @@
 # more of a shell scripty thing.
 
 # Get configuration settings
-. ../config.out
+. ../config.mk
 
 if [ $1 ] ; then PREFIX=$1 ; fi
 PROBE=../cardmgr/pcic_probe
@@ -138,14 +138,8 @@
     fi
 }
 
-if [ "$SYSV_INIT" = "y" ] ; then
-    install_sysv
-else
-    install_bsd
-fi
 
 install_clients
-install_depmod
 
 # Directory to receive cardmgr's 'stab' socket status file
 mkdir -p $PREFIX/var/lib/pcmcia
diff -Nru pcmcia-cs-3.2.7/include/pcmcia/autoconf.h pcmcia-cs-3.2.7-new/include/pcmcia/autoconf.h
--- pcmcia-cs-3.2.7/include/pcmcia/autoconf.h	1970-01-01 01:00:00.000000000 +0100
+++ pcmcia-cs-3.2.7-new/include/pcmcia/autoconf.h	2004-05-01 17:37:16.591559232 +0200
@@ -0,0 +1 @@
+/* Dummy */
diff -Nru pcmcia-cs-3.2.7/man/Makefile pcmcia-cs-3.2.7-new/man/Makefile
--- pcmcia-cs-3.2.7/man/Makefile	2003-09-15 07:14:41.000000000 +0200
+++ pcmcia-cs-3.2.7-new/man/Makefile	2004-05-01 18:04:16.812248248 +0200
@@ -16,9 +16,7 @@
 	memory_cb.4 ibmtr_cs.4 smc91c92_cs.4 iflash2_mtd.4 iflash2+_mtd.4 \
 	wvlan_cs.4 ray_cs.4 airo.4 airo_cs.4 axnet_cs.4 dummy_cb.4
 MAN5 = pcmcia.5 stab.5
-MAN8 = cardmgr.8 cardctl.8 ftl_format.8 ftl_check.8 scsi_info.8 \
-	ifport.8 ifuser.8 ide_info.8 pcinitrd.8 pack_cis.8 dump_cis.8 \
-	lspnp.8 setpnp.8 pcic_probe.8
+MAN8 = cardmgr.8 cardctl.8 pack_cis.8 dump_cis.8
 
 all: $(MAN1_X11) $(MAN4) $(MAN5) $(MAN8)
 
@@ -42,9 +40,9 @@
 
 install-man8: $(MAN8)
 	@mkdir -p $(MANDIR)/man8
-	cp *.8 $(MANDIR)/man8
+	cp $(MAN8) $(MANDIR)/man8
 
-install: install-man1-x11 install-man4 install-man5 install-man8
+install: install-man5 install-man8
 
 man2html: $(MAN1) $(MAN4) $(MAN5) $(MAN8)
 	for f in *.[1458] ; do				\
