--- vnc_unixsrc/vncserver	2007-04-26 12:57:06.000000000 +0200
+++ vnc_unixsrc-new/vncserver	2007-05-08 14:29:16.000000000 +0200
@@ -36,13 +36,13 @@
 $geometry = "1024x768";
 $depth = 24;
 $desktopName = "X";
-$vncClasses = "/usr/local/vnc/classes";
+$vncClasses = "/usr/share/vnc/classes";
 $vncUserDir = "$ENV{HOME}/.vnc";
-$fontPath = "unix/:7100";
+$fontPath = "/usr/X11R6/lib/X11/fonts/misc,/usr/X11R6/lib/X11/fonts/Type1,/usr/X11R6/lib/X11/fonts/75dpi";
 $authType = "-rfbauth $vncUserDir/passwd";
 
 # Here is another example of setting the font path:
-# $fontPath = "/usr/lib/X11/fonts/misc/,/usr/lib/X11/fonts/75dpi/";
+# $fontPath = "$fontPath = "unix/:7100";
 
 # X colors database path is optional, uncomment and edit to use:
 # $colorPath = "/usr/lib/X11/rgb";
