--- vncserver~	2003-07-31 08:19:37.000000000 -0600
+++ vncserver	2005-07-17 14:34:38.000000000 -0600
@@ -36,12 +36,12 @@
 $geometry = "1024x768";
 $depth = 24;
 $desktopName = "X";
-$vncClasses = "/usr/local/vnc/classes";
+$vncClasses = "/usr/share/vnc/classes";
 $vncUserDir = "$ENV{HOME}/.vnc";
-$fontPath = "unix/:7100";
+$fontPath = "/usr/X11R6/lib/X11/fonts/misc,/usr/X11R6/lib/X11/fonts/Type1,/usr/X11R6/lib/X11/fonts/75dpi";
 
 # Here is another example of setting the font path:
-# $fontPath = "/usr/lib/X11/fonts/misc/,/usr/lib/X11/fonts/75dpi/";
+# $fontPath = "$fontPath = "unix/:7100";
 
 # X colors database path is optional, uncomment and edit to use:
 # $colorPath = "/usr/lib/X11/rgb";
