diff -Nru xc-old/extras/drm/shared/drm.h xc/extras/drm/shared/drm.h
--- xc-old/extras/drm/shared/drm.h	2006-01-09 14:58:53.000000000 +0100
+++ xc/extras/drm/shared/drm.h	2006-01-09 15:00:49.000000000 +0100
@@ -63,9 +63,6 @@
 #endif
 
 #if defined(__linux__)
-#if defined(__KERNEL__)
-#include <linux/config.h>
-#endif
 #include <asm/ioctl.h>		/* For _IO* macros */
 #define DRM_IOCTL_NR(n)		_IOC_NR(n)
 #define DRM_IOC_VOID		_IOC_NONE
