diff -Nru dovecot-1.0.rc2.orig/dovecot-example.conf dovecot-1.0.rc2/dovecot-example.conf
--- dovecot-1.0.rc2.orig/dovecot-example.conf	2006-07-04 20:13:50.000000000 +0200
+++ dovecot-1.0.rc2/dovecot-example.conf	2006-07-04 20:13:59.000000000 +0200
@@ -13,10 +13,10 @@
 # --with-ssldir=/etc/ssl
 
 # Base directory where to store runtime data.
-#base_dir = /var/run/dovecot/
+base_dir = /var/run/dovecot/
 
 # Protocols we want to be serving: imap imaps pop3 pop3s
-#protocols = imap imaps
+protocols = imaps pop3s
 
 # IP or host address where to listen in for connections. It's not currently
 # possible to specify multiple addresses. "*" listens in all IPv4 interfaces.
@@ -86,8 +86,8 @@
 # dropping root privileges, so keep the key file unreadable by anyone but
 # root. Included doc/mkcert.sh can be used to easily generate self-signed
 # certificate, just make sure to update the domains in dovecot-openssl.cnf
-#ssl_cert_file = /etc/ssl/certs/dovecot.pem
-#ssl_key_file = /etc/ssl/private/dovecot.pem
+ssl_cert_file = /etc/ssl/certs/dovecot.crt
+ssl_key_file = /etc/ssl/keys/dovecot.key
 
 # If key file is password protected, give the password here. Alternatively
 # give it when starting dovecot with -p parameter.
@@ -118,7 +118,7 @@
 # which login needs to be able to connect to. The sockets are created when
 # running as root, so you don't have to worry about permissions. Note that
 # everything in this directory is deleted when Dovecot is started.
-#login_dir = /var/run/dovecot/login
+login_dir = /var/run/dovecot/login
 
 # chroot login process to the login_dir. Only reason not to do this is if you
 # wish to run the whole Dovecot without roots.
@@ -130,7 +130,7 @@
 # only it has access, it's used to control access for authentication process.
 # Note that this user is NOT used to access mails.
 # http://wiki.dovecot.org/UserIds
-#login_user = dovecot
+login_user = dovecot
 
 # Set max. process size in megabytes. If you don't use
 # login_process_per_connection you might need to grow this.
@@ -140,11 +140,11 @@
 # login process be allowed to process multiple connections (no)? Yes is more
 # secure, espcially with SSL/TLS enabled. No is faster since there's no need
 # to create processes all the time.
-#login_process_per_connection = yes
+login_process_per_connection = no
 
 # Number of login processes to create. If login_process_per_connection is
 # yes, this is the number of extra processes waiting for users to log in.
-#login_processes_count = 3
+login_processes_count = 1
 
 # Maximum number of extra login processes to create. The extra process count
 # usually stays at login_processes_count, but when multiple users start logging
@@ -192,7 +192,7 @@
 #   default_mail_env = mbox:~/mail/:INBOX=/var/mail/%u
 #   default_mail_env = mbox:/var/mail/%d/%n/:INDEX=/var/indexes/%d/%n
 #
-#default_mail_env = 
+default_mail_env = mbox:~/Mail:INBOX=/var/spool/mail/%u
 
 # If you need to set multiple mailbox locations or want to change default
 # namespace settings, you can do it by defining namespace sections:
@@ -294,7 +294,7 @@
 # non-valid GID as primary group ID aren't allowed to log in. If user
 # belongs to supplementary groups with non-valid GIDs, those groups are
 # not set.
-#first_valid_gid = 1
+first_valid_gid = 100
 #last_valid_gid = 0
 
 # Maximum number of running mail processes. When this limit is reached,
@@ -467,19 +467,19 @@
 
 protocol imap {
   # Login executable location.
-  #login_executable = /usr/libexec/dovecot/imap-login
+  login_executable = /usr/lib/dovecot/imap-login
 
   # IMAP executable location. Changing this allows you to execute other
   # binaries before the imap process is executed.
   #
   # This would write rawlogs into ~/dovecot.rawlog/ directory:
-  #   mail_executable = /usr/libexec/dovecot/rawlog /usr/libexec/dovecot/imap
+  #   mail_executable = /usr/lib/dovecot/rawlog /usr/lib/dovecot/imap
   #
   # This would attach gdb into the imap process and write backtraces into
   # /tmp/gdbhelper.* files:
-  #   mail_executable = /usr/libexec/dovecot/gdbhelper /usr/libexec/dovecot/imap
+  #   mail_executable = /usr/lib/dovecot/gdbhelper /usr/lib/dovecot/imap
   #
-  #mail_executable = /usr/libexec/dovecot/imap
+  mail_executable = /usr/lib/dovecot/imap
 
   # Maximum IMAP command line length in bytes. Some clients generate very long
   # command lines with huge mailboxes, so you may need to raise this if you get
@@ -529,10 +529,10 @@
 
 protocol pop3 {
   # Login executable location.
-  #login_executable = /usr/libexec/dovecot/pop3-login
+  login_executable = /usr/lib/dovecot/pop3-login
 
   # POP3 executable location
-  #mail_executable = /usr/libexec/dovecot/pop3
+  mail_executable = /usr/lib/dovecot/pop3
 
   # Don't try to set mails non-recent or seen with POP3 sessions. This is
   # mostly intended to reduce disk I/O. With maildir it doesn't move files
@@ -576,7 +576,7 @@
   # installations. %08Xu%08Xv will be the new default, so use it for new
   # installations.
   #
-  #pop3_uidl_format = 
+  pop3_uidl_format = %08Xu%08Xv
 
   # POP3 logout format string:
   #  %t - number of TOP commands
@@ -622,7 +622,7 @@
   #mail_plugin_dir = /usr/lib/dovecot/lda
 
   # Binary to use for sending mails.
-  #sendmail_path = /usr/lib/sendmail
+  #sendmail_path = /usr/sbin/sendmail
 
   # UNIX socket path to master authentication server to find users.
   #auth_socket_path = /var/run/dovecot/auth-master
@@ -633,7 +633,7 @@
 ##
 
 # Executable location
-#auth_executable = /usr/libexec/dovecot/dovecot-auth
+auth_executable = /usr/lib/dovecot/dovecot-auth
 
 # Set max. process size in megabytes.
 #auth_process_size = 256
@@ -745,7 +745,7 @@
   # database (passwd usually), you can use static userdb.
   # REMEMBER: You'll need /etc/pam.d/dovecot file created for PAM
   # authentication to actually work.
-  passdb pam {
+  #passdb pam {
     #  [session=yes] [setcred=yes] [cache_key=<key>] [<service name>]
     #
     # session=yes makes Dovecot open and immediately close PAM session. Some
@@ -773,7 +773,7 @@
     #   args = session=yes *
     #   args = cache_key=%u dovecot
     #args = dovecot
-  }
+  #}
 
   # /etc/passwd or similar, using getpwnam()
   # In many systems nowadays this uses Name Service Switch, which is
@@ -782,8 +782,8 @@
   #}
 
   # /etc/shadow or similiar, using getspnam(). Deprecated by PAM nowadays.
-  #passdb shadow {
-  #}
+  passdb shadow {
+  }
 
   # BSD authentication. Used by at least OpenBSD.
   #passdb bsdauth {
@@ -885,7 +885,7 @@
   # authentication with BSDs internally accesses shadow files, which also
   # requires roots. Note that this user is NOT used to access mails.
   # That user is specified by userdb above.
-  user = root
+  #user = root
 
   # Directory where to chroot the process. Most authentication backends don't
   # work if this is set, and there's no point chrooting if auth_user is root.
