diff -urN mjpegtools-1.6.2.orig/configure mjpegtools-1.6.2/configure
--- mjpegtools-1.6.2.orig/configure	2004-02-03 16:58:37.000000000 -0600
+++ mjpegtools-1.6.2/configure	2005-05-10 09:50:33.646553048 -0500
@@ -25858,16 +25858,16 @@
   fi
   have_cmov=false
   case $arch in
-  i786) ARCHFLAGS="-mcpu=i786 -march=$arch"
+  i786) ARCHFLAGS="-march=$arch"
   have_cmov=true
   ;;
-  i686) ARCHFLAGS="-mcpu=i686 -march=$arch"
+  i686) ARCHFLAGS="-march=$arch"
   have_cmov=true
   ;;
-  i586) ARCHFLAGS="-mcpu=i586 -march=$arch"
+  i586) ARCHFLAGS="-march=$arch"
   have_cmov=false
   ;;
-  *) ARCHFLAGS="-mcpu=i686 -march=$arch"
+  *) ARCHFLAGS="-march=$arch"
   have_cmov=true
   ;;
   esac
diff -urN mjpegtools-1.6.2.orig/lavtools/lavpipe.c mjpegtools-1.6.2/lavtools/lavpipe.c
--- mjpegtools-1.6.2.orig/lavtools/lavpipe.c	2002-09-23 14:46:43.000000000 -0500
+++ mjpegtools-1.6.2/lavtools/lavpipe.c	2005-05-10 09:50:55.723196888 -0500
@@ -582,6 +582,7 @@
       mjpeg_info( "closing input %d (source %d)", i, current_index);
       decommission_pipe_source(source);
     KEEP_SOURCE:
+		;
     }
   }
 }
diff -urN mjpegtools-1.6.2.orig/mpeg2enc/picture.cc mjpegtools-1.6.2/mpeg2enc/picture.cc
--- mjpegtools-1.6.2.orig/mpeg2enc/picture.cc	2003-10-25 12:36:32.000000000 -0500
+++ mjpegtools-1.6.2/mpeg2enc/picture.cc	2005-05-10 09:51:19.622563632 -0500
@@ -83,9 +83,9 @@
     }
 
 
-	curref = new (uint8_t *)[5];
-	curorg = new (uint8_t *)[5];
-	pred   = new (uint8_t *)[5];
+	curref = new uint8_t *[5];
+	curorg = new uint8_t *[5];
+	pred   = new uint8_t *[5];
 
 	for( i = 0 ; i<3; i++)
 	{
