diff -aur libmusicbrainz-2.1.5.orig/lib/c_wrapper.cpp libmusicbrainz-2.1.5/lib/c_wrapper.cpp
--- libmusicbrainz-2.1.5.orig/lib/c_wrapper.cpp	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1.5/lib/c_wrapper.cpp	2008-06-29 18:17:54.484557560 +0200
@@ -24,6 +24,7 @@
 #include "musicbrainz.h"
 #include "trm.h"
 #include "mb_c.h"
+#include <cstring>
 
 extern "C"
 {
diff -aur libmusicbrainz-2.1.5.orig/lib/comsocket.h libmusicbrainz-2.1.5/lib/comsocket.h
--- libmusicbrainz-2.1.5.orig/lib/comsocket.h	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1.5/lib/comsocket.h	2008-06-29 18:17:34.844561280 +0200
@@ -34,6 +34,7 @@
 #define COMSOCKET_H
 #include <stdio.h>
 #include <string>
+#include <cstring>
 
 #ifdef WIN32
 #include <winsock.h>
diff -aur libmusicbrainz-2.1.5.orig/lib/http.cpp libmusicbrainz-2.1.5/lib/http.cpp
--- libmusicbrainz-2.1.5.orig/lib/http.cpp	2007-05-11 08:21:21.000000000 +0200
+++ libmusicbrainz-2.1.5/lib/http.cpp	2008-06-29 18:18:14.562308600 +0200
@@ -27,6 +27,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <ctype.h>
+#include <cstring>
 #ifndef WIN32
 #include <sys/time.h>
 #endif
diff -aur libmusicbrainz-2.1.5.orig/lib/sigclient.h libmusicbrainz-2.1.5/lib/sigclient.h
--- libmusicbrainz-2.1.5.orig/lib/sigclient.h	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1.5/lib/sigclient.h	2008-06-29 18:18:32.262560320 +0200
@@ -33,6 +33,7 @@
 
 #include <iostream>
 #include <string>
+#include <cstring>
 #include <map>
 #include <vector>
 
