diff -aur quilt-0.46.orig/Makefile.in quilt-0.46/Makefile.in
--- quilt-0.46.orig/Makefile.in	2006-10-11 14:36:10.000000000 +0200
+++ quilt-0.46/Makefile.in	2006-10-26 17:57:47.322105535 +0200
@@ -12,7 +12,7 @@
 docdir :=	@docdir@
 mandir :=	@mandir@
 localedir :=	$(datadir)/locale
-etcdir :=	$(subst /usr/etc,/etc,$(prefix)/etc)
+etcdir :=	$(prefix)/etc
 
 INSTALL :=	@INSTALL@
 POD2MAN :=	@POD2MAN@
@@ -325,10 +325,10 @@
 	$(INSTALL) -d $(BUILD_ROOT)$(mandir)/man1
 	$(INSTALL) -m 644 $(MAN1) $(BUILD_ROOT)$(mandir)/man1/
 
+	$(INSTALL) -d $(BUILD_ROOT)$(libdir)
+	$(INSTALL) -d $(BUILD_ROOT)$(libdir)/bash-completion
+	$(INSTALL) -m 644 bash_completion $(BUILD_ROOT)$(libdir)/bash-completion/quilt
 	$(INSTALL) -d $(BUILD_ROOT)$(etcdir)
-	$(INSTALL) -d $(BUILD_ROOT)$(etcdir)/bash_completion.d
-	$(INSTALL) -m 644 bash_completion				\
-		   $(BUILD_ROOT)$(etcdir)/bash_completion.d/quilt
 	$(INSTALL) -m 644 quilt.quiltrc $(BUILD_ROOT)$(etcdir)/
 
 install-compat: install-compat1 $(COMPAT_SYMLINKS:%=install-compat-symlink-%)
diff -aur quilt-0.43.orig/bin/quilt.in quilt-0.43/bin/quilt.in
--- quilt-0.43.orig/bin/quilt.in	2006-01-16 20:33:00.000000000 +0100
+++ quilt-0.43/bin/quilt.in	2006-02-03 23:00:24.000000000 +0100
@@ -14,7 +14,7 @@
 
 if [ -z "$QUILTRC" ]
 then
-	for QUILTRC in $HOME/.quiltrc /etc/quilt.quiltrc; do
+	for QUILTRC in $HOME/.quiltrc /usr/etc/quilt.quiltrc; do
 		[ -e $QUILTRC ] && break
 	done
 	export QUILTRC
