diff -aur gdb-6.8.orig/gdb/eval.c gdb-6.8/gdb/eval.c
--- gdb-6.8.orig/gdb/eval.c	2008-02-04 01:23:04.000000000 +0100
+++ gdb-6.8/gdb/eval.c	2009-06-11 16:05:46.018550469 +0200
@@ -1647,6 +1647,8 @@
 	struct type *tmp_type;
 	int offset_item;	/* The array offset where the item lives */
 
+	memset (subscript_array, 0, sizeof (subscript_array));
+
 	if (nargs > MAX_FORTRAN_DIMS)
 	  error (_("Too many subscripts for F77 (%d Max)"), MAX_FORTRAN_DIMS);
 
