diff -urN mplayer-1.0_rc2_p28058/libavcodec/libx264.c mplayer-1.0_rc2_p28058-new/libavcodec/libx264.c
--- mplayer-1.0_rc2_p28058/libavcodec/libx264.c	2008-12-02 13:56:20.000000000 +1100
+++ mplayer-1.0_rc2_p28058-new/libavcodec/libx264.c	2008-12-03 00:54:35.824823293 +1100
@@ -164,7 +164,7 @@
 
     x4->params.i_bframe = avctx->max_b_frames;
     x4->params.b_cabac = avctx->coder_type == FF_CODER_TYPE_AC;
-    x4->params.i_bframe_adaptive = avctx->b_frame_strategy;
+    x4->params.b_bframe_adaptive = avctx->b_frame_strategy;
     x4->params.i_bframe_bias = avctx->bframebias;
     x4->params.b_bframe_pyramid = avctx->flags2 & CODEC_FLAG2_BPYRAMID;
     avctx->has_b_frames= avctx->flags2 & CODEC_FLAG2_BPYRAMID ? 2 : !!avctx->max_b_frames;
