diff -Nru mpg123.orig/Makefile mpg123/Makefile
--- mpg123.orig/Makefile	2006-06-16 07:13:41.000000000 +0200
+++ mpg123/Makefile	2006-06-16 07:28:22.000000000 +0200
@@ -226,11 +226,9 @@
 linux-esd:
 	$(MAKE) CC=gcc LDFLAGS= \
 		AUDIO_LIB='-lesd -laudiofile' \
-		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
-			audio_esd.o' \
+		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o audio_esd.o term.o' \
 		CFLAGS='$(CFLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
-			-DOSS -DUSE_ESD \
-			-Wall  -O2 -m486 \
+			-DOSS -DUSE_ESD -DTERM_CONTROL -Wall \
 			-fomit-frame-pointer -funroll-all-loops \
 			-finline-functions -ffast-math \
 			$(RPM_OPT_FLAGS)' \
diff -Nru mpg123.orig/audio.c mpg123/audio.c
--- mpg123.orig/audio.c	2006-06-16 07:13:41.000000000 +0200
+++ mpg123/audio.c	2006-06-16 07:27:30.000000000 +0200
@@ -1,4 +1,4 @@
-
+#include <stdlib.h>
 #include "mpg123.h"
 
 void audio_info_struct_init(struct audio_info_struct *ai)
diff -Nru mpg123.orig/audio_nas.c mpg123/audio_nas.c
--- mpg123.orig/audio_nas.c	2006-06-16 07:13:41.000000000 +0200
+++ mpg123/audio_nas.c	2006-06-16 07:27:30.000000000 +0200
@@ -185,7 +185,7 @@
     info.buf_size = buf_samples * ai->channels * AuSizeofFormat(format);
     info.buf = (char *) malloc(info.buf_size);
     if (info.buf == NULL) {
-        fprintf(stderr, "Unable to allocate input/output buffer of size %ld\n",
+        fprintf(stderr, "Unable to allocate input/output buffer of size %d\n",
              info.buf_size);
         exit(1);
     }
