diff -Nru irssi-0.8.10.orig/src/irc/dcc/dcc.c irssi-0.8.10/src/irc/dcc/dcc.c
--- irssi-0.8.10.orig/src/irc/dcc/dcc.c	2005-12-08 19:32:48.000000000 +0200
+++ irssi-0.8.10/src/irc/dcc/dcc.c	2006-04-03 01:47:57.000000000 +0300
@@ -58,8 +58,8 @@
 
 	pos = gslist_find_string(dcc_types, type);
 	if (pos != NULL) {
-                dcc_types = g_slist_remove(dcc_types, pos->data);
 		g_free(pos->data);
+                dcc_types = g_slist_remove(dcc_types, pos->data);
 	}
 }
 
