diff -Nru libtunepimp-0.3.0/configure libtunepimp-0.3.0-new/configure
--- libtunepimp-0.3.0/configure	2004-03-30 11:42:24.000000000 +0200
+++ libtunepimp-0.3.0-new/configure	2004-09-06 17:05:51.348758754 +0200
@@ -19591,7 +19591,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lreadline -ltermcap $LIBS"
+LIBS="-lreadline -lncurses $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -19650,7 +19650,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_readline_readline" >&5
 echo "${ECHO_T}$ac_cv_lib_readline_readline" >&6
 if test $ac_cv_lib_readline_readline = yes; then
-  READLINE_LIBS="-lreadline -ltermcap"
+  READLINE_LIBS="-lreadline -lncurses"
 
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_LIBREADLINE 1
diff -Nru libtunepimp-0.3.0/configure.in libtunepimp-0.3.0-new/configure.in
--- libtunepimp-0.3.0/configure.in	2004-03-30 11:40:04.000000000 +0200
+++ libtunepimp-0.3.0-new/configure.in	2004-09-06 17:05:51.349758217 +0200
@@ -46,14 +46,14 @@
 
 dnl Check for readline. This is optional.
 AC_CHECK_LIB(readline, readline,
-             [READLINE_LIBS="-lreadline -ltermcap"
+             [READLINE_LIBS="-lreadline -lncurses"
              AC_DEFINE(HAVE_LIBREADLINE,1,[Readline Support])],
              [echo "*"
              echo "*  The GNU readline library isn't installed."
              echo "*  This means the sample tagger, tp_tagger, won't"
              echo "*  have readline support."
              echo "*"
-	     ],-ltermcap)
+	     ],-lncurses)
 
 dnl Check for MAD decoder lib for mp3 support
 AC_CHECK_LIB(mad,mad_version,
diff -Nru libtunepimp-0.3.0/lib/threads/posix/thread.h libtunepimp-0.3.0-new/lib/threads/posix/thread.h
--- libtunepimp-0.3.0/lib/threads/posix/thread.h	2004-03-25 10:20:15.000000000 +0100
+++ libtunepimp-0.3.0-new/lib/threads/posix/thread.h	2004-09-06 17:29:30.996443199 +0200
@@ -28,6 +28,7 @@
 #define __THREAD_H_
 
 #include <pthread.h>
+#include <signal.h>
 #include "semaphore.h"
 
 #include "defs.h"
