diff -ur ../libtunepimp-0.4.1.orig/configure ./configure
--- ../libtunepimp-0.4.1.orig/configure	2006-03-29 00:15:50.000000000 -0700
+++ ./configure	2006-03-29 00:17:22.000000000 -0700
@@ -20228,7 +20228,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lreadline -ltermcap $LIBS"
+LIBS="-lreadline -lncurses $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
