Index: trunk/configure.ac
===================================================================
--- trunk/configure.ac (revision 32)
+++ trunk/configure.ac (revision 39)
@@ -1670,10 +1670,7 @@
 		khidemap="${enableval}"
 	   fi
-	   sh_syscalltable=`grep 'D sys_call_table' ${khidemap} | awk '{print $1}'`
+	   sh_syscalltable=`egrep '(D|d|R|r) sys_call_table' ${khidemap} | awk '{print $1}'`
 	   if test x"$sh_syscalltable" = x; then
-		sh_syscalltable=`grep 'd sys_call_table' ${khidemap} | awk '{print $1}'`
-		if test x"$sh_syscalltable" = x; then
-			AC_MSG_ERROR([--enable-khide: symbol sys_call_table not found in ${khidemap}])
-		fi
+		AC_MSG_ERROR([--enable-khide: symbol sys_call_table not found in ${khidemap}])
 	   fi
 	   sh_syscalltable="0x${sh_syscalltable}"
