diff -Nru pkg-get-0.4.5.orig/scripts/pkg-get.pl pkg-get-0.4.5/scripts/pkg-get.pl
--- pkg-get-0.4.5.orig/scripts/pkg-get.pl	2006-07-13 05:04:11.000000000 +0200
+++ pkg-get-0.4.5/scripts/pkg-get.pl	2006-07-17 13:01:37.524719500 +0200
@@ -943,7 +943,7 @@
 						printf("%-19s %-19s %-19s\n\n","Package","Installed","Available in the repositories");
 						$gotdiff = 1;
 					}
-					printf("%-19s %-19s %-19s %-19s\n", $pkg{'name'},  $pkg{'instversion'}, $pkg{'version'}."-".$pkg{'release'});
+					printf("%-19s %-19s %-19s\n", $pkg{'name'},  $pkg{'instversion'}, $pkg{'version'}."-".$pkg{'release'});
 				} elsif ($all) { # yeah, it blows, at least avoid to read the locked state twice.
 					if ($gotdiff == 0){
 						print "Differences between installed packages and packages repo:\n\n";
