diff -ruN shared-mime-info-0.14.orig/configure shared-mime-info-0.15/configure
--- shared-mime-info-0.14.orig/configure	2004-04-26 13:40:35.884056512 -0400
+++ shared-mime-info-0.15/configure	2004-04-26 13:40:36.098023984 -0400
@@ -3513,7 +3513,7 @@
 
     fi
   fi
-     USE_NLS=yes
+     USE_NLS=no
 
 
     gt_cv_have_gettext=no
@@ -4440,7 +4440,7 @@
     if test "$gt_cv_have_gettext" = "yes" ; then
 
 cat >>confdefs.h <<\_ACEOF
-#define ENABLE_NLS 1
+#define ENABLE_NLS 0
 _ACEOF
 
     fi
