diff -Nru tgif-QPL-4.1.45.orig/Tgif.tmpl tgif-QPL-4.1.45/Tgif.tmpl
--- tgif-QPL-4.1.45.orig/Tgif.tmpl	2006-10-25 21:37:30.000000000 +0200
+++ tgif-QPL-4.1.45/Tgif.tmpl	2006-10-25 21:39:49.000000000 +0200
@@ -38,9 +38,9 @@
 TGIFDIR		= $(LIBDIR)/tgif
 XCOMM CDEBUGFLAGS= -g -Wall
 XCOMM INSTPGMFLAGS= -s
-XCOMM BINDIR	= /u/halfmoon/bin
-XCOMM MANPATH	= /u/halfmoon/man
-XCOMM TGIFDIR	= /u/halfmoon/lib/X11/tgif
+BINDIR	= /usr/bin
+MANPATH	= /usr/man
+TGIFDIR	= /usr/share/tgif
 XCOMM MOREDEFINES	= -D_HAS_STREAMS_SUPPORT -DENABLE_NLS -D_TGIF_DBG
 LOCALEDIR	= /usr/share/locale
 DIRDEFINES	= -DTGIF_PATH=\"$(TGIFDIR)\" -DPSFILE_MOD=0664 \@@\
@@ -56,7 +56,7 @@
                   -D_USE_PS_ADOBE_STRING=\"3.0/3.0\" -D_DONT_USE_MKTEMP \@@\
 		  -D_DONT_REENCODE=\"FFDingbests:ZapfDingbats\" \@@\
 		  -DUSE_XT_INITIALIZE
-MOREDEFINES	=
+MOREDEFINES	= -DA4PAPER
 
 XCOMM ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 XCOMM Please see Imakefile for more defines to be added to MOREDEFINES above
