diff --git a/libcap/include/sys/capability.h b/libcap/include/sys/capability.h
index 4b1f7a3..d27017d 100644
--- a/libcap/include/sys/capability.h
+++ b/libcap/include/sys/capability.h
@@ -19,24 +19,12 @@ extern "C" {
  * information for the user library.
  */
 
+/* Linux headers are now sanitized properly for userspace */
+#include <linux/capability.h>
 #include <sys/types.h>
 #include <stdint.h>
 
 /*
- * Make sure we can be included from userland by preventing
- * capability.h from including other kernel headers
- */
-#define _LINUX_TYPES_H
-#define _LINUX_FS_H
-#define __LINUX_COMPILER_H
-#define __user
-
-typedef unsigned int __u32;
-typedef __u32 __le32;
-
-#include <linux/capability.h>
-
-/*
  * POSIX capability types
  */
 
