--- xxdiff-3.2/src/app.cpp.orig	2006-08-04 16:40:25.144936920 +0200
+++ xxdiff-3.2/src/app.cpp	2006-08-04 16:40:46.151743400 +0200
@@ -265,22 +265,6 @@
    // Read in the resources and create resources object.
    _resources = buildResources();
 
-#ifndef XX_KDE
-#if (QT_VERSION >= 0x030000)
-   // By default, if not specified, force SGI style.
-   if ( _cmdline._forceStyle == false ) {
-      _style = QStyleFactory::create( _resources->getStyleKey() );
-      setStyle( _style );
-   }
-#endif
-#endif
-
-#ifndef XX_KDE
-   if ( _cmdline._forceFont == false ) {
-      setFont( _resources->getFontApp(), true );
-   }
-#endif
-
    if ( _cmdline._promptForFiles ) {
       promptForFiles( _cmdline );
    }
