--- xxdiff-cvs-diff.old	2004-10-29 10:40:21.000000000 +0200
+++ xxdiff-cvs-diff	2004-10-29 21:32:00.103934168 +0200
@@ -115,7 +115,7 @@
         # a temporary file.
         tmpf = NamedTemporaryFile(prefix=tmppfx)
         cin, cout = os.popen2(
-            'patch --reverse --output "%s"' % tmpf.name, 'rw')
+            'patch -p0 --reverse --output "%s"' % tmpf.name, 'rw')
         cin.write('Index: %s\n' % filename)
         cin.write(patch)
         cin.close()
