diff -aur ruby-gtk2-0.14.1.orig/gtk/src/rbgdkpixbuf.c ruby-gtk2-0.14.1/gtk/src/rbgdkpixbuf.c
--- ruby-gtk2-0.14.1.orig/gtk/src/rbgdkpixbuf.c	2005-11-14 09:14:07.000000000 +0100
+++ ruby-gtk2-0.14.1/gtk/src/rbgdkpixbuf.c	2006-01-29 11:34:19.000000000 +0100
@@ -73,7 +73,7 @@
     GdkPixmap *pixmap;
     GdkBitmap *mask;
 
-    rb_scan_args(argc, argv, "01", &colormap_or_alpha, &alpha);
+    rb_scan_args(argc, argv, "11", &colormap_or_alpha, &alpha);
 
     if (rb_obj_is_kind_of(colormap_or_alpha, GTYPE2CLASS(GDK_TYPE_COLORMAP))){
         gdk_pixbuf_render_pixmap_and_mask_for_colormap(_SELF(self),
diff -aur ruby-gtk2-0.14.1.orig/gtk/src/rbgtkwidget.c ruby-gtk2-0.14.1/gtk/src/rbgtkwidget.c
--- ruby-gtk2-0.14.1.orig/gtk/src/rbgtkwidget.c	2005-11-14 09:14:07.000000000 +0100
+++ ruby-gtk2-0.14.1/gtk/src/rbgtkwidget.c	2006-01-29 11:40:12.000000000 +0100
@@ -340,7 +340,7 @@
 widget_set_colormap(self, colormap)
     VALUE self, colormap;
 {
-    gtk_widget_set_colormap(_SELF(self), GDK_COLORMAP(RVAL2GOBJ(self)));
+    gtk_widget_set_colormap(_SELF(self), GDK_COLORMAP(RVAL2GOBJ(colormap)));
     return self;
 }
 
