diff -Nru mathopd-1.5p6.orig/src/request.c mathopd-1.5p6/src/request.c
--- mathopd-1.5p6.orig/src/request.c	2009-08-24 16:32:23.822054573 +0200
+++ mathopd-1.5p6/src/request.c	2009-08-24 16:34:53.672062355 +0200
@@ -210,7 +210,7 @@
 	return buf;
 }
 
-static char *getline(struct pool *p, int fold)
+static char *get_line(struct pool *p, int fold)
 {
 	char *s, *olds, *sp, *end;
 	int f;
@@ -247,7 +247,7 @@
 			break;
 		}
 	}
-	log_d("getline: fallen off the end");
+	log_d("get_line: fallen off the end");
 	return 0;
 }
 
@@ -1172,7 +1172,7 @@
 	unsigned long cl;
 
 	do {
-		l = getline(&r->cn->header_input, 0);
+		l = get_line(&r->cn->header_input, 0);
 		if (l == 0)
 			return -1;
 	} while (*l == 0);
@@ -1201,7 +1201,7 @@
 	if (r->protocol_major && r->protocol_minor)
 		r->cn->keepalive = 1;
 	n = 0;
-	while ((l = getline(&r->cn->header_input, 1)) != 0) {
+	while ((l = get_line(&r->cn->header_input, 1)) != 0) {
 		s = strchr(l, ':');
 		if (s == 0)
 			continue;
