diff -aur epdfview-0.1.6.orig/src/IDocumentLink.h epdfview-0.1.6/src/IDocumentLink.h
--- epdfview-0.1.6.orig/src/IDocumentLink.h	2006-10-17 16:23:19.000000000 +0200
+++ epdfview-0.1.6/src/IDocumentLink.h	2008-10-25 12:33:20.478624280 +0200
@@ -33,7 +33,7 @@
     class IDocumentLink
     {
         public:
-            IDocumentLink (gdouble x1, gdouble y1, gdouble x2, gdouble y1);
+            IDocumentLink (gdouble x1, gdouble y1, gdouble x2, gdouble y2);
             virtual ~IDocumentLink (void);
 
             virtual void activate (IDocument *document) = 0;
