diff -Nru aterm-1.0.0.orig/autoconf/configure.in aterm-1.0.0/autoconf/configure.in
--- aterm-1.0.0.orig/autoconf/configure.in	2006-07-12 07:11:18.000000000 +0200
+++ aterm-1.0.0/autoconf/configure.in	2006-07-12 07:12:39.000000000 +0200
@@ -352,7 +352,7 @@
 			AC_MSG_RESULT($AFTERSTEP_LIBS)
 		else
 	    	    if test "$afterstep_config" != "NO" ; then 	
-		   ÿÿÿÿ	AC_MSG_RESULT(afterstep-config not found in $afterstep_config)	
+			AC_MSG_RESULT(afterstep-config not found in $afterstep_config)	
 		    else
 			AC_MSG_RESULT(afterstep-config not found)	 
 		    fi
