diff -Nru stunnel-4.14.orig/configure stunnel-4.14/configure
--- stunnel-4.14.orig/configure	2005-11-03 08:01:22.000000000 +0100
+++ stunnel-4.14/configure	2005-11-03 08:10:51.000000000 +0100
@@ -23339,7 +23339,7 @@
 
 { echo "$as_me:$LINENO: **************************************** write the results" >&5
 echo "$as_me: **************************************** write the results" >&6;}
-CPPFLAGS="$CPPFLAGS -DLIBDIR=\\\"$libdir\\\" -DCONFDIR=\\\"$sysconfdir/stunnel\\\" -DPIDFILE=\\\"$localstatedir/stunnel/stunnel.pid\\\""
+CPPFLAGS="$CPPFLAGS -DLIBDIR=\\\"$libdir\\\" -DCONFDIR=\\\"$sysconfdir\\\" -DPIDFILE=\\\"$localstatedir/run/stunnel/stunnel.pid\\\""
                                                                       ac_config_files="$ac_config_files Makefile src/Makefile src/stunnel3 doc/Makefile tools/Makefile tools/stunnel.conf-sample tools/stunnel.init"
 
 cat >confcache <<\_ACEOF
diff -Nru stunnel-4.14.orig/tools/stunnel.conf-sample.in stunnel-4.14/tools/stunnel.conf-sample.in
--- stunnel-4.14.orig/tools/stunnel.conf-sample.in	2005-11-03 08:01:22.000000000 +0100
+++ stunnel-4.14/tools/stunnel.conf-sample.in	2005-11-03 08:01:40.000000000 +0100
@@ -3,13 +3,13 @@
 ; Please make sure you understand them (especially the effect of chroot jail)
 
 ; Certificate/key is needed in server mode and optional in client mode
-cert = @prefix@/etc/stunnel/mail.pem
-;key = @prefix@/etc/stunnel/mail.pem
+cert = /etc/ssl/certs/stunnel.crt
+key = /etc/ssl/keys/stunnel.key
 
 ; Some security enhancements for UNIX systems - comment them out on Win32
-chroot = @prefix@/var/stunnel/
+chroot = /var/run/stunnel/
 setuid = nobody
-setgid = nogroup
+setgid = nobody
 ; PID is created inside chroot jail
 pid = /stunnel.pid
 
@@ -36,7 +36,7 @@
 
 ; Some debugging stuff useful for troubleshooting
 ;debug = 7
-;output = stunnel.log
+;output = /var/log/stunnel.log
 
 ; Use it for client mode
 ;client = yes
