diff -Nru mutt-1.5.15.orig/muttlib.c mutt-1.5.15/muttlib.c
--- mutt-1.5.15.orig/muttlib.c	2007-05-27 19:17:55.000000000 +0200
+++ mutt-1.5.15/muttlib.c	2007-05-27 19:26:39.000000000 +0200
@@ -540,7 +540,7 @@
     if (dest[idx] == '&')
     {
       memmove (&dest[idx + pwnl], &dest[idx + 1],
-	       MAX(destlen - idx - pwnl - 1, 0));
+	       MAX((ssize_t)(destlen - idx - pwnl - 1), 0));
       memcpy (&dest[idx], pw->pw_name, MIN(destlen - idx - 1, pwnl));
       dest[idx] = toupper ((unsigned char) dest[idx]);
     }
