diff -Nru chroot_safe-1.4.orig/Makefile.in chroot_safe-1.4/Makefile.in
--- chroot_safe-1.4.orig/Makefile.in	2005-04-09 12:22:27.000000000 +0300
+++ chroot_safe-1.4/Makefile.in	2005-04-11 08:16:30.000000000 +0300
@@ -2,9 +2,9 @@
 CXX=g++
 SOFLAGS=-fpic -shared
 DEFINES=
-prefix=@prefix@
+prefix=$(DESTDIR)@prefix@
 exec_prefix=@exec_prefix@
-libexec=@libexecdir@/chroot_safe
+libexec=$(DESTDIR)@libexecdir@/chroot_safe
 sbin=@sbindir@
 man=@mandir@/man1
 install=@INSTALL@
