Index: saslauthd/auth_shadow.c
--- saslauthd/auth_shadow.c	2005-11-30 21:49:20.000000000 +0100
+++ saslauthd/auth_shadow.c	2005-11-30 21:49:41.000000000 +0100
@@ -45,6 +45,7 @@
 # include <time.h>
 # include <pwd.h>
 # include <syslog.h>
+# include <crypt.h>
 # ifndef HAVE_GETSPNAM
 
 # ifdef WITH_DES

Index: lib/server.c
--- lib/server.c   2004-11-17 15:19:30 +0100
+++ lib/server.c   2005-05-26 20:54:36 +0200
@@ -473,16 +473,8 @@
       return SASL_CONTINUE;
   }
 
-  /* get the path to the plugins; for now the config file will reside there */
-  getpath_cb=_sasl_find_getpath_callback( global_callbacks.callbacks );
-  if (getpath_cb==NULL) return SASL_BADPARAM;
-
-  /* getpath_cb->proc MUST be a sasl_getpath_t; if only c had a type
-     system */
-  result = ((sasl_getpath_t *)(getpath_cb->proc))(getpath_cb->context,
-						  &path_to_config);
-  if (result!=SASL_OK) goto done;
-  if (path_to_config == NULL) path_to_config = "";
+  result = SASL_OK;
+  path_to_config = "/etc/sasl/apps";
 
   c = strchr(path_to_config, PATHS_DELIMITER);
 
