--- acroread.orig	2006-08-04 01:45:40.000000000 +0200
+++ acroread	2006-08-04 01:47:39.000000000 +0200
@@ -455,13 +455,13 @@
             continue
         fi
 
-        len_version=`expr length $version`
-        base_version=`expr substr "${base_version}0000000000" 1 $len_version`
-        len_base_version=`expr length $base_version`
+        len_version=`expr length "$version"`
+        base_version=`expr substr "${base_version}0000000000" 1 "$len_version"`
+        len_base_version=`expr length "$base_version"`
 
         while [ $len_version -gt $len_base_version ]; do
-            base_version=`expr substr "${base_version}0000000000" 1 $len_version`
-            len_base_version=`expr length $base_version`
+            base_version=`expr substr "${base_version}0000000000" 1 "$len_version"`
+            len_base_version=`expr length "$base_version"`
         done
 
     	if [ $version -ge $base_version ]; then
