diff -Nru exim-4.66.orig/src/tls-openssl.c exim-4.66/src/tls-openssl.c
--- exim-4.66.orig/src/tls-openssl.c	2007-02-28 20:33:35.527246099 +0000
+++ exim-4.66/src/tls-openssl.c	2007-02-28 20:34:47.751361917 +0000
@@ -343,8 +343,11 @@
 /* Set up the information callback, which outputs if debugging is at a suitable
 level. */
 
-if (!(SSL_CTX_set_info_callback(ctx, (void (*)())info_callback)))
-  return tls_error(US"SSL_CTX_set_info_callback", host);
+/* if (!(SSL_CTX_set_info_callback(ctx, (void (*)())info_callback)))
+  return tls_error(US"SSL_CTX_set_info_callback", host); */
+
+SSL_CTX_set_info_callback(ctx, (void (*)())info_callback);
+ 
 
 /* The following patch was supplied by Robert Roselius */
 
