--- ipprequest.cpp.orig	2005-10-10 17:06:30.000000000 +0200
+++ ipprequest.cpp	2006-05-14 21:32:00.000000000 +0200
@@ -511,6 +511,11 @@
 	cupsFreeOptions(n, options);
 
 	// find an remove that annoying "document-format" attribute
+	#if CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2
+	ipp_attribute_t *attr = ippFindAttribute(request_, "document-format", IPP_TAG_NAME);
+	ippDeleteAttribute(request_, attr);
+	#else
+	// (can't use IppDeleteAttribute as older cups doesn't have that)
 	ipp_attribute_t	*attr = request_->attrs;
 	while (attr)
 	{
@@ -523,4 +528,5 @@
 		}
 		attr = attr->next;
 	}
+	#endif
 }
