diff -Nru kdelibs-3.5.1-old/configure kdelibs-3.5.1/configure
--- kdelibs-3.5.1-old/configure	2006-02-04 23:33:36.000000000 +0100
+++ kdelibs-3.5.1/configure	2006-02-04 23:36:12.000000000 +0100
@@ -51445,6 +51445,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
+#include <pcre.h>
 #include <pcreposix.h>
 int
 main ()
diff -Nru kdelibs-3.5.1-old/configure.in kdelibs-3.5.1/configure.in
--- kdelibs-3.5.1-old/configure.in	2006-02-04 23:33:37.000000000 +0100
+++ kdelibs-3.5.1/configure.in	2006-02-04 23:36:51.000000000 +0100
@@ -1948,7 +1948,8 @@
       ac_LDFLAGS_save="$LDFLAGS"
       LDFLAGS="$LDFLAGS $all_libraries"
       AC_TRY_LINK(
-         [#include <pcreposix.h>],
+         [#include <pcre.h>
+#include <pcreposix.h>],
          [regfree(0);],
          [ac_cv_have_pcreposix="yes"],
          [ac_cv_have_pcreposix="no"]
diff -Nru kdelibs-3.5.1-old/kjs/configure.in.in kdelibs-3.5.1/kjs/configure.in.in
--- kdelibs-3.5.1-old/kjs/configure.in.in	2006-02-04 23:33:32.000000000 +0100
+++ kdelibs-3.5.1/kjs/configure.in.in	2006-02-04 23:37:35.000000000 +0100
@@ -38,7 +38,8 @@
       ac_LDFLAGS_save="$LDFLAGS"
       LDFLAGS="$LDFLAGS $all_libraries"
       AC_TRY_LINK(
-         [#include <pcreposix.h>],
+         [#include <pcre.h>
+#include <pcreposix.h>],
          [regfree(0);],
          [ac_cv_have_pcreposix="yes"],
          [ac_cv_have_pcreposix="no"]
