diff -Nru evilwm-1.0.1.orig/screen.c evilwm-1.0.1/screen.c
--- evilwm-1.0.1.orig/screen.c	2009-03-30 13:53:24.000000000 +0200
+++ evilwm-1.0.1/screen.c	2009-03-30 13:54:00.000000000 +0200
@@ -317,8 +317,8 @@
 		} else {
 			c->oldx = c->x;
 			c->oldw = c->width;
-			c->x = 0;
-			c->width = DisplayWidth(dpy, c->screen->screen);
+			c->x = 1;
+			c->width = DisplayWidth(dpy, c->screen->screen) - 2;
 		}
 	}
 	if (hv & MAXIMISE_VERT) {
@@ -329,8 +329,8 @@
 		} else {
 			c->oldy = c->y;
 			c->oldh = c->height;
-			c->y = 0;
-			c->height = DisplayHeight(dpy, c->screen->screen);
+			c->y = 1;
+			c->height = DisplayHeight(dpy, c->screen->screen) - 2;
 		}
 	}
 	moveresize(c);
