diff -Nru evilwm-0.99.24.orig/Makefile evilwm-0.99.24/Makefile
--- evilwm-0.99.24.orig/Makefile	2006-02-20 19:02:37.000000000 +0100
+++ evilwm-0.99.24/Makefile	2006-02-20 19:02:48.000000000 +0100
@@ -63,7 +63,7 @@
 #DEFINES += -DXDEBUG	# show some X calls
 
 DEFINES += -DVERSION=\"$(version)\" $(DEBIAN)
-CFLAGS  += $(INCLUDES) $(DEFINES) -Os -Wall
+CFLAGS  += $(INCLUDES) $(DEFINES) -Wall
 #CFLAGS  += $(INCLUDES) $(DEFINES) -g -Wall
 CFLAGS  += -W -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wshadow -Waggregate-return -Wnested-externs -Winline -Wwrite-strings -Wundef -Wsign-compare -Wmissing-prototypes -Wredundant-decls
 LDFLAGS += $(LDPATH) $(LIBS)
@@ -84,9 +84,9 @@
 
 install: evilwm
 	if [ -f evilwm.exe ]; then mv evilwm.exe evilwm; fi
-	mkdir -p $(prefix)/bin $(prefix)/share/man/man1
+	mkdir -p $(prefix)/bin $(prefix)/man/man1
 	install -s evilwm $(prefix)/bin
-	install evilwm.1 $(prefix)/share/man/man1
+	install evilwm.1 $(prefix)/man/man1
 	#gzip -9 $(prefix)/share/man/man1/evilwm.1
 
 dist:
diff -Nru evilwm-0.99.24.orig/evilwm.1 evilwm-0.99.24/evilwm.1
--- evilwm-0.99.24.orig/evilwm.1	2006-02-20 19:02:37.000000000 +0100
+++ evilwm-0.99.24/evilwm.1	2006-02-20 19:02:48.000000000 +0100
@@ -70,7 +70,7 @@
 bottom-right of screen.
 .IP I
 Show information about current window.
-.IP Equals
+.IP V
 Maximise current window vertically (toggle).
 .IP X
 Maximise current window (toggle).
diff -Nru evilwm-0.99.24.orig/keymap.h evilwm-0.99.24/keymap.h
--- evilwm-0.99.24.orig/keymap.h	2006-02-20 19:02:37.000000000 +0100
+++ evilwm-0.99.24/keymap.h	2006-02-20 19:02:48.000000000 +0100
@@ -14,7 +14,7 @@
 #define KEY_LOWER	XK_Insert
 #define KEY_ALTLOWER	XK_KP_Insert
 #define KEY_INFO	XK_i
-#define KEY_MAXVERT	XK_equal
+#define KEY_MAXVERT	XK_v
 #define KEY_MAX		XK_x
 #define KEY_FIX		XK_f
 #define KEY_PREVDESK	XK_Left
