diff -C 3 -r gegl-0.0.18-original/operations/external/ff-load.c gegl-0.0.18/operations/external/ff-load.c
*** gegl-0.0.18-original/operations/external/ff-load.c	Sat Jun 14 18:40:04 2008
--- gegl-0.0.18/operations/external/ff-load.c	Thu Oct  2 20:35:34 2008
***************
*** 27,33 ****
  
  #include "gegl-chant.h"
  #include <errno.h>
! #include <ffmpeg/avformat.h>
  
  typedef struct
  {
--- 27,33 ----
  
  #include "gegl-chant.h"
  #include <errno.h>
! #include <libavformat/avformat.h>
  
  typedef struct
  {
